webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleViewHandler.java

```diff
@@ -244,7 +242,8 @@ protected void wrappedService(HttpServletRequest request, HttpServletResponse re
                 sendBytes(request, response, totalPdfOutput, "application/pdf");
             } else if (extension.equals("zip")) {
                 ByteArrayOutputStream zipOutput = ScrambleRequest
-                        .requestsToZip(getServletContext(), globalTitle, generationDate, scrambleRequests, password, generationUrl);
+                        .requestsToZip(getServletContext(), globalTitle, generationDate, query);
```

Comment:
Similar comment here that I think the query parsing should happen up here in the `*Handler` class, rather than down in `ScrambleRequest.requestsToZip`

Reply:
Fixed after our video chat